tms/src/main/java/com/netcracker/edu/tms/ui/ProjectRestController.java

```diff
@@ -40,13 +46,26 @@
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
 
+        //this block must be before last return line, because in case of invalid addProject function block must not be executed
```

Comment:
As i see comments about operations ordering
But Email notification goes before logic of adding new project to database
And in case of
`boolean operationResult = projectService.addProject(newProj);`
'operationResult == false'
we will notify about failed operation